api/app/controllers/spree/api/v2/storefront/taxons_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class TaxonsController < ::Spree::Api::V2::BaseController
+          def index
+            @taxons = Spree::Taxon.includes(:children).accessible_by(current_ability, :read)
+            @taxons = dependencies[:taxons_finder].new(@taxons, params).execute
+
+            render json: serialize_taxons(@taxons), status: 200
+          end
+
+          def show
+            @taxon = Spree::Taxon.find_by(id: params[:id])
```

Comment:
@VygovskySergey we need to use friendly id here - `Spree::Taxon.friendly.find(params[:id])` - this will allow us to find by ID or slug (which is a must)

Reply:
@damianlegawiec what do you think about `Spree::Taxon.where(id: params[:id]).or(Spree::Taxon.where(slug: params[:id]))` and starting to slowly move away from friendly_id dependency? Project is barely alive and I remember I had to wait a lot of time for their update to rails 5.2(even after they got PR with code ready to merge)

Reply:
And should we support finding by database id or maybe just keep with permalink?

Reply:
@bbonislawski it won't work, slugs aren't kept in the Taxons table, also people heavily use history feature which is crucial for stuff like Taxon which change quite often

Reply:
@bbonislawski update - there is a permalink field - let's use it :)

Reply:
```
Spree::Taxon.find_by(id: params[:id]).or(Spree::Taxon.find_by(permalink: params[:id]))
```